### PR TITLE
Gqw/add qr and sign

### PR DESCRIPTION
**Pull request by Gong-air**
PPO算子增加

### CODE REVIEW

File: diopi_test/python/conformance/diopi_configs.py [hunk 1]
```diff
@@ -4812,6 +4812,47 @@
...
+                    "ins": ['input'],
+                    "shape": ((1024, 64), (384, 128),
+                              (64, 1, 128), (128, 64, 3, 3),
+                              (2, 32, 130, 130)),
+                },
+            ],
+        ),
+    ),
+
+    'linalgqrout': dict(
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4836]: 都不加out

File: diopi_test/python/conformance/gen_data.py [hunk 2]
```diff
@@ -492,6 +492,15 @@ def ctc_loss(log_probs, targets, input_lengths, target_lengths, blank=0, reducti
         loss = torch.nn.functional.ctc_loss(log_probs_, targets, input_lengths, target_lengths, blank=blank, reduction=reduction, zero_infinity=zero_infinity)
         return loss
 
+    def linalgqrout(input, mode):
+        q, r = torch.linalg.qr(input, mode)
+        out = [q, r]
+        return out
+
+    def amaxout(input, dim, keepdim):
```
Comment (xintian-514) [inline diopi_test/python/conformance/gen_data.py:500]: 这个可以不用？

File: proto/include/diopi/functions.h [hunk 3]
```diff
@@ -2214,6 +2214,10 @@ DIOPI_API diopiError_t diopiSgnInp(diopiContextHandle_t ctx, diopiTensorHandle_t
  */
 DIOPI_API diopiError_t diopiIsNan(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t input);
 
+DIOPI_API diopiError_t diopiLinalgQR(diopiContextHandle_t ctx, diopiConstTensorHandle_t A, const char* mode, diopiTensorHandle_t Q, diopiTensorHandle_t R);
```
Comment (xintian-514) [inline proto/include/diopi/functions.h:2217]: 注释顺便加了吧

File: diopi_test/python/conformance/diopi_configs.py [hunk 4]
```diff
@@ -4812,6 +4812,47 @@
         ),
     ),
 
+    'amaxout': dict(
+        name=['amaxout'],
+        interface=['CustomizedTest'],
+        dtype=[Dtype.float64, Dtype.float32, Dtype.float16, Dtype.int16, Dtype.int32, Dtype.int64, Dtype.int8, Dtype.uint8],
+        para=dict(
+            dim=[1, 0, 2, (1, 2), (1, 2)],
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4820]: dim可以是None吗

File: diopi_test/python/conformance/diopi_configs.py [hunk 5]
```diff
@@ -4812,6 +4812,47 @@
...
+        dtype=[Dtype.float64, Dtype.float32],
+        para=dict(
+            mode=['reduced', 'reduced', 'reduced', 'reduced', 'reduced'],
+        ),
+        tensor_para=dict(
+            gen_fn=Genfunc.randn,
+            args=[
+                {
+                    "ins": ['input'],
+                    "shape": ((1024, 1024), (384, 384),
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4848]: 加一个dim[-2]>dim[-1]的测例吧

File: diopi_test/python/conformance/diopi_functions.py [hunk 6]
```diff
@@ -3733,3 +3733,70 @@ def isnan(input) -> Tensor:
...
     check_returncode(ret)
     return out
+
+
+def amaxout(input, dim, keepdim) -> Tensor:
+    call = "diopiAmax"
+    func = check_function(call)
+    if isinstance(dim, int):
+        sizeO = []
+        # import pdb
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_functions.py:3743]: 删掉

File: diopi_test/python/conformance/diopi_functions.py [hunk 7]
```diff
@@ -3733,3 +3733,70 @@ def isnan(input) -> Tensor:
...
+        # pdb.set_trace()
+        dim_num = dim
+        dim = Sizes(list([dim]))
+        sizeI = list(input.size().data)
+        if dim_num < 0 and abs(dim_num) > len(list(input.shape().data)):
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_functions.py:3748]: if dim_num < 0:
    if abs(dim_num) > len(list(input.shape().data)):
        ...
    else:
        ...

File: diopi_test/python/conformance/diopi_configs.py [hunk 8]
```diff
@@ -4812,6 +4812,47 @@
...
+            ],
+        ),
+    ),
+
+    'linalgqrout': dict(
+        name=['linalgqrout'],
+        interface=['CustomizedTest'],
+        dtype=[Dtype.float64, Dtype.float32],
+        para=dict(
+            mode=['reduced', 'reduced', 'reduced', 'reduced', 'reduced'],
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4841]: mode可以是其他值么

File: proto/include/diopi/functions.h [hunk 9]
```diff
@@ -2214,6 +2214,10 @@ DIOPI_API diopiError_t diopiSgnInp(diopiContextHandle_t ctx, diopiTensorHandle_t
...
+
+DIOPI_API diopiError_t diopiAmax(diopiContextHandle_t ctx, diopiTensorHandle_t out, diopiConstTensorHandle_t self, diopiSize_t dim, bool keepdim);
```
Comment (xintian-514) [inline proto/include/diopi/functions.h:2219]: 这个dim可以为空么

File: diopi_test/python/conformance/diopi_functions.py [hunk 10]
```diff
@@ -3733,3 +3733,70 @@ def isnan(input) -> Tensor:
     ret = func(input.context(), out, input)
     check_returncode(ret)
     return out
+
+
+def amaxout(input, dim, keepdim) -> Tensor:
+    call = "diopiAmax"
+    func = check_function(call)
+    if isinstance(dim, int):
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_functions.py:3741]: 企业微信截图_16893060651640
求out的形状研究研究这个函数能不能用

File: diopi_test/python/conformance/diopi_configs.py [hunk 11]
```diff
@@ -4812,6 +4812,47 @@
...
+                              (2, 32, 130, 130)),
+                },
+            ],
+        ),
+    ),
+
+    'linalgqrout': dict(
+        name=['linalgqrout'],
+        interface=['CustomizedTest'],
+        dtype=[Dtype.float64, Dtype.float32],
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4839]: dtype只能是这两个类型吗

Comment (Gong-air) [reply diopi_test/python/conformance/diopi_configs.py:4839]: 是的，其他的pytorch会报错

File: diopi_test/python/conformance/diopi_configs.py [hunk 12]
```diff
@@ -4812,6 +4812,47 @@
...
+        para=dict(
+            mode=['reduced', 'reduced', 'complete', 'complete', 'r', 'r'],
+        ),
+        tensor_para=dict(
+            gen_fn=Genfunc.randn,
+            args=[
+                {
+                    "ins": ['input'],
+                    "shape": ((1024, 384), (384, 1024),
```
Comment (xintian-514) [inline diopi_test/python/conformance/diopi_configs.py:4848]: 后两维大小一样的也保留吧

**Review by xintian-514 — APPROVED** [review]
